path_login.go

```diff
@@ -136,6 +136,19 @@ func (b *azureAuthBackend) pathLogin(ctx context.Context, req *logical.Request,
 	vmName := data.Get("vm_name").(string)
 	resourceID := data.Get("resource_id").(string)
 
+	if !validateAzureField(subscriptionID) {
+		return logical.ErrorResponse(fmt.Sprintf("invalid subscription id %q", subscriptionID)), nil
+	}
+	if !validateAzureField(resourceGroupName) {
```

Comment:
It looks like the regex is too restrictive based on the allowed characters listed here for resourceGroups: https://learn.microsoft.com/en-us/azure/azure-resource-manager/management/resource-name-rules#microsoftresources